src/NuGet.Core/NuGet.Commands/RestoreCommand/CompatibilityChecker.cs

```diff
@@ -60,22 +61,33 @@ public CompatibilityChecker(IReadOnlyList<NuGetv3LocalRepository> localRepositor
             // * The Targets (TxMs) defined in the project.json, with no Runtimes
             // * All combinations of TxMs and Runtimes defined in the project.json
             // * Additional (TxMs, Runtime) pairs defined by the "supports" mechanism in project.json
-
             var runtimeAssemblies = new HashSet<string>(StringComparer.OrdinalIgnoreCase);
             var compileAssemblies = new Dictionary<string, LibraryIdentity>(StringComparer.OrdinalIgnoreCase);
             var issues = new List<CompatibilityIssue>();
 
+            if (packageSpec.RestoreMetadata?.ProjectStyle == ProjectStyle.DotnetToolReference)
+            {
+                if (packageSpec.GetAllPackageDependencies().Count() != 1)
```

Comment:
can this return a null list for zero package dependencies?

Reply:
No, it'll return an empty set.